tests/accounts/test_views.py

```diff
@@ -167,3 +263,27 @@ def test_post_forgets_user(self, monkeypatch, pyramid_request):
         assert result.headers["foo"] == "bar"
         assert forget.calls == [pretend.call(pyramid_request)]
         assert pyramid_request.session.invalidate.calls == [pretend.call()]
+
+    def test_post_redirects_user(self, monkeypatch, pyramid_request):
+        forget = pretend.call_recorder(lambda request: [("foo", "bar")])
+        monkeypatch.setattr(views, "forget", forget)
+
+        # The set of all possible next URLs. Since this set is infinite, we
+        # test only a finite set of reasonable URLs.
+        nexts = ["/security/"]
+
+        pyramid_request.method = "POST"
+        pyramid_request.session = pretend.stub(
+            invalidate=pretend.call_recorder(lambda: None),
+        )
+
+        for next in nexts:
```

Comment:
This doesn't actually test the failure case at all, I would make this something like:

```python

@pytest.mark.parametrize(
    ("next_url", "expected"),
    [
        ("/security/", "/security/"),
        ("http://example.com/", "/"),
    ],
)
def test_post_redirects_user(self, monkeypatch, pyramid_request, next_url, expected):
    ....
    pyramid_request.POST["next"] = next_url
    ...
    assert result.headers["Location"] == expected
```